### PR TITLE
Add QLineF support

### PR DESCRIPTION
**Pull request by Montel**

### CODE REVIEW

File: crates/cxx-qt-lib/src/core/qlinef.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib/qlinef.h"
+
+#include "../assertion_utils.h"
+
+#include <cstdint>
+
+// QLine has "QPoint" members - pt1, pt2
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/core/qlinef.cpp:13]: this is QLineF and they are doubles not 32bit integers and the links need updating :-)

File: crates/cxx-qt-lib/src/core/qlinef.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+impl Default for QLineF {
+    /// Constructs a default qlinef
+    fn default() -> Self {
+        ffi::qlinef_default()
+    }
+}
+
+// Safety:
+//
+// Static checks on the C++ side ensure that QLine is trivial.
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/core/qlinef.rs:108]: QLineF

File: crates/cxx-qt-lib/src/core/qlinef.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        #[rust_name = "qlinef_default"]
+        fn construct() -> QLineF;
+
+        #[doc(hidden)]
+        #[rust_name = "qlinef_new"]
+        fn construct(pt1: QPointF, pt2: QPointF) -> QLineF;
+    }
+}
+
+/// The QLine class provides a two-dimensional vector using integer precision
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/core/qlinef.rs:85]: QLineF

File: crates/cxx-qt-lib/src/core/qlinef.rs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+#[derive(Debug, Clone, PartialEq)]
+#[repr(C)]
+pub struct QLineF {
+    pt1: QPointF,
+    pt2: QPointF,
+}
+
+impl QLineF {
+    pub fn new(pt1: QPointF, pt2: QPointF) -> Self {
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/core/qlinef.rs:94]: add a doc line here

**Comment by ahayzen-kdab** [comment]
Overall looks good, small comments :-)

File: crates/cxx-qt-lib/src/core/qlinef.cpp (hunk 1 above)
Comment (Montel) [reply crates/cxx-qt-lib/src/core/qlinef.cpp:13]: link is ok as qline and qlineF is in same file.

File: tests/qt_types_standalone/rust/src/qlinef.rs [hunk 5]
```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx_qt_lib::QLineF;
```
Comment (ahayzen-kdab) [inline tests/qt_types_standalone/rust/src/qlinef.rs:6]: could be `use cxx_qt_lib::{QLineF, QPointF}`

**Review by ahayzen-kdab — APPROVED** [review]
LGTM

**Pull request merged**